### PR TITLE
Ability to Edit a Post

### LINKED ISSUE #21: IMPLEMENT: Ability to edit a post
This is an often requested feature by mods, but I think would be well served to others too.

@andysbolton did quite a bit of work on the service side for this, and thus far has it so that people can edit a post they made in the last ten minutes (which is I think fair).

On the front end, I've got some of this wired up as a POC, to test on @andysbolton 's changes.

### PR DESCRIPTION
**Pull request by DarthHater**
This is a largely WIP POC of editing posts.

So far it just wires up a few things, and brings in a library for inline editing.

This would eventually fix #21

### CODE REVIEW

File: src/auth/authentication.js [hunk 1]
```diff
@@ -7,6 +7,13 @@ export function isLoggedIn() {
     return true;
 }
 
+export function checkUser(userId) {
+    if (userId == getUserId()) {
```
Comment (DarthHater) [inline src/auth/authentication.js]: Is there a difference between `==` and `===` for this?

Comment (DarthHater) [reply src/auth/authentication.js]: I think `==` does a type conversion, if there is no need we could do `===`

File: src/components/Login/Login.js [hunk 2]
```diff
@@ -75,7 +75,7 @@ class Login extends Component {
                 <Grid item xs={12} sm={6} style={{textAlign: 'center'}}>
                     <h4>Enter your credentials to login and start shit posting</h4>
                     {this.props.user.error &&
-                        <FormHelperText error>{this.props.user.error}</FormHelperText>
+                        <FormHelperText error style={{textAlign: 'center'}}>{this.props.user.error}</FormHelperText>
```
Comment (DarthHater) [inline src/components/Login/Login.js]: Probably worth putting this into `styles` and importing it from that.

File: src/helpers/PostHelper.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export function canEditPost(post) {
+    let milliseconds = new Date().getTime() - new Date(post.PostedAt).getTime();
+    return (milliseconds / 60000) < 10;
+}
```
Comment (DarthHater) [inline src/helpers/PostHelper.js]: Trivial: Newline

File: src/auth/authentication.js (hunk 1 above)
Comment (andysbolton) [reply src/auth/authentication.js]: Oh true, I'll switch that over

File: src/styles/utilities.scss [hunk 4]
```diff
@@ -0,0 +1,7 @@
+.w-100 {
+    width: 100% !important;
+}
+
+.text-center {
+    text-align: center !important;
+}
```
Comment (DarthHater) [inline src/styles/utilities.scss]: Trivial: Newline

File: src/components/Login/Login.js (hunk 2 above)
Comment (DarthHater) [reply src/components/Login/Login.js]: What I meant was you can do add it to:

https://github.com/DarthHater/bored-board-webapp/blob/928c02aadb6691eede00c760796f3d42dbf18deb/src/components/Login/Login.js#L27

like:

```
    santaDog: {
        textAlign: 'center'
    },
    formHelper: {
       textAlign: 'center'
    }
```

And import like `style={styles.formHelper}`

This keeps the style isolated to the class, which is kinda nice.

**Pull request merged**